test/local-fs.spec.ts

```diff
@@ -114,6 +115,14 @@ describe(`the local filesystem implementation`, () => {
                 .then(() => expect(fs.loadTextFile(fileName)).to.eventually.equals(newContent));
         });
 
+        it(`ignores events from ignored dir in watcher`, () => {
```

Comment:
Perhaps we need a new implementation `class FakeFs extends InMemoryFs` that behaves exactly like the memory FS only it also has ability to turn event emitting on and off, and manually emit events including the error event (to simulate scenarios where a fileSystem losing sync with underlying source of data)